compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -66,21 +66,15 @@ impl<'a> ArchiveBuilder<'a> for LlvmArchiveBuilder<'a> {
         archive: &Path,
         skip: Box<dyn FnMut(&str) -> bool + 'static>,
     ) -> io::Result<()> {
-        let mut archive = archive.to_path_buf();
-        if self.sess.target.llvm_target.contains("-apple-macosx") {
-            if let Some(new_archive) = try_extract_macho_fat_archive(&self.sess, &archive)? {
-                archive = new_archive
-            }
-        }
```

Comment:
Maybe we should keep this for now in case we need to revert back from the rust rewrite to the original version on macOS due to a bug in the rust rewrite?

Reply:
The LlvmArchiveWriter doesn't have a clean place to add this -- it doesn't track file_range, it just has a path.

Reply:
I see. Would you mind waiting a week or so to make sure no issues pop up? r=me after that.

Reply:
yes, that's fine